plugins/builds/steps/update.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+
+module.exports = () => ({
+    method: 'PUT',
+    path: '/builds/{id}/steps/{name}',
+    config: {
+        description: 'Update a step for a build',
+        notes: 'Writes a step record',
+        tags: ['api', 'builds', 'steps'],
+        auth: {
+            strategies: ['token'],
+            scope: ['build']
+        },
+        handler: (request, reply) => {
+            const factory = request.server.app.buildFactory;
+            const buildId = request.params.id;
+            const buildIdCred = request.auth.credentials.username;
+            let stepIndex = -1;
+
+            if (buildId !== buildIdCred) {
+                throw boom.forbidden(`Credential only valid for ${buildIdCred}`);
+            }
+
+            return factory.get(buildId)
+                .then(build => {
+                    if (!build) {
+                        throw boom.notFound(`Build ${buildId} does not exist`);
+                    }
+                    const steps = build.steps;
+
+                    stepIndex = steps.reduce((prevIndex, step, index) => {
```

Comment:
Same thing in one line:
`steps.findIndex(s => s.name === request.params.name);`